### ISSUE TITLE
Class Diagram for the Project

### ISSUE THREAD

**Issue by hakanaktas0** [bounswe/bounswe2023group9#37]
### Issue Description

Class Diagram for the project should be prepared. Classes include the following;

### Steps

- [x] #38 
- [x] #39
- [x] #40  
- [x] #41
- [x] #42 
- [x] #43 
- [x] #44 
- [x] #45 
- [x] #46 
- [x] #47 
- [x] #56 
- [x] #48 
- [x] #49 
- [x] #50 
- [x] #51 
- [x] #52 
- [x] #53 
- [x] #54 
- [x] #55 

### Deadline for the Issue

07.04.2023

**Comment by Bera0422** [comment]
Class diagrams for Node, Question, Theorem and Proof Classes were created and need review. They can also be seen on this [link:](https://lucid.app/lucidchart/68361e85-695a-4c99-b073-3bf5da13b35c/edit?invitationId=inv_f7c380fa-3aed-452b-8bf6-2687b31241ec&page=r.VLewbugIZd#)
![image](https://user-images.githubusercontent.com/69896238/230484885-5989f621-3855-4d55-a1c0-d81c2f879c3b.png)

**Comment by hakanaktas0** [comment]
All class diagrams have been completed by the team and merged by @Zulalm and @hakanaktas0.

**Comment by uyduranomer** [comment]
I have reviewed the design and combining. There was a missing link between annotation and node classes. I added it. Also I have edited the design for aesthetical purposes. There was no other problem with the design. The last version can be found in the class diagrams page of the lucidchart workspace and here in this pdf:
[Class Diagrams - Class Diagrams.pdf](https://github.com/bounswe/bounswe2023group9/files/11191030/Class.Diagrams.-.Class.Diagrams.pdf)
